### ISSUE TITLE
PHP 8.2.0 Deprecated: Creation of dynamic property WP_CLI\Dispatcher\Subcommand::$longdesc is deprecated in phar:///………/wp-cli/php/WP_CLI/Dispatcher/CompositeCommand.php

### ISSUE THREAD

**Issue by porg** [wp-cli/wp-cli#5713]
## On macOS 11.7.1 I now have wp-cli 2.7.1 with PHP 8.2.0

```
$ php --version
PHP 8.2.0 (cli) (built: Dec  9 2022 17:05:41) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.0, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.0, Copyright (c), by Zend Technologies

$ cat ~/.wp-cli/config.yml

# Global parameter defaults
path: ~/Documents # Real path but no WP there or anywhere locally.

# Aliases to remote installs
@mydomain:
  ssh: customerID@customerID-ssh.services.mywebhost.com:10959
  path: /data/web/customerID/html/apps/mydomain/
```

## Connecting to remote wp-cli 2.7.1 with PHP 8.0.22 on Linux 4.19.0

```
$ wp @mydomain --info
```

👇 The following lines repeat dozens/hundreds of times:

```
PHP Deprecated:  Creation of dynamic property WP_CLI\Dispatcher\Subcommand::$longdesc is deprecated in phar:///usr/local/Cellar/wp-cli/2.7.1/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CompositeCommand.php on line 38

Deprecated: Creation of dynamic property WP_CLI\Dispatcher\Subcommand::$longdesc is deprecated in phar:///usr/local/Cellar/wp-cli/2.7.1/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CompositeCommand.php on line 38
```

👇 Before I get the actual output from the remote wp-cli 2.7.1 which uses PHP 8.0.22

```
OS:	Linux 4.19.0-22-amd64 #1 SMP Debian 4.19.260-1 (2022-09-29) x86_64
Shell:	/bin/bash
PHP binary:	/usr/bin/php8.0
PHP version:	8.0.22
php.ini used:	/etc/php/8.0/cli/php.ini
MySQL binary:	/usr/bin/mysql
MySQL version:	mysql  Ver 14.14 Distrib 5.7.39-42, for debian-linux-gnu (x86_64) using  7.0
SQL modes:	
WP-CLI root dir:	phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/data/web/customerID
WP-CLI packages dir:	/data/web/customerID/.wp-cli/packages/
WP-CLI cache dir:	/data/web/customerID/.wp-cli/cache
WP-CLI global config:	/data/web/customerID/.wp-cli/config.yml
WP-CLI project config:	
WP-CLI version:	2.7.1
```

**Comment by porg** [comment]
After I [finally mastered to set up my remote wp-cli](https://github.com/wp-cli/wp-cli/issues/5245#issuecomment-1349814049) for a few times today I ran my remote wp-cli just fine.

Then I did a `brew upgrade` the first time in months, and that upgraded PHP to 8.2.0 and since then I got this deprecation output on each p-cli command.

`brew info php` confirms that 8.2.0 is fresh from today:
==> php: stable 8.2.0 (bottled), HEAD
/usr/local/Cellar/php/8.2.0 (520 files, 81.8MB) *
  Poured from bottle on **2022-12-15 at 14:55:45**  👈 Just from this afternoon.

- Do I need to downgrade PHP with brew?  
- Or is there any way to suppress the deprecation output via php configuration? (while still working correctly with wp-cli!)

**Comment by porg** [comment]
```
$ wp @mydomain db check

Tons of deprecated lines.

And then the command itself also fails:

mysqlcheck: Got error: 2002: Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2) when trying to connect
```
👆 No idea whether this is a genuine "wp db" bug, or a consequence of the "Deprecated:" lines.



```
wp @dd db export > backup.sql
```
👆 Those "Deprecated: …" lines are in the STDOUT and hence also in the output file.

**Comment by porg** [comment]
A day later I discovered that each failed attempt also left an empty (seemingly) temporary file in the home directory on my webhost:

```
-rw-r--r-- 1 customerid  www-data    0 Dec 16 00:57 realDbName-backup-969a531.sql
-rw-r--r-- 1 customerid  www-data    0 Dec 16 01:03 realDbName-backup-9af1875.sql
-rw-r--r-- 1 customerid  www-data    0 Dec 16 01:03 realDbName-backup-e33af9d.sql
```

**Comment by danielbachhuber** [comment]
@porg Do you see the notices with the nightly build? They should've been fixed with https://github.com/wp-cli/wp-cli/pull/5707

You can use `wp cli update --nightly` to install it.

**Comment by porg** [comment]
```
$ wp cli update --nightly
Error: /usr/local/Cellar/wp-cli/2.7.1/bin/wp is not writable by current user.

$ brew uninstall wp-cli  
Uninstalling /usr/local/Cellar/wp-cli/2.7.1... (3 files, 6.4MB)

# Installing via curl one-liner as ~/bin/wp

$ wp cli update --nightly

... Numerous Deprecation lines ...

You have version 2.7.1. Would you like to update to the latest nightly? [y/n]

yes ENTER

Downloading from https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli-nightly.phar...

... Numerous Deprecation lines ...

md5 hash verified: 918e7c6d31ea52bc1adbd577a6a8f84e

|

```

From there on the prompt remains stuck:
- Network activity only the first few seconds (downloading). Then nothing.
- No CPU load.
- All "php" processes in Activity Monitor idle at 0%.

Terminating the open process with ctrl-C.
- Tried 3 times. Each time the same.

Any ideas?

**Comment by danielbachhuber** [comment]
It's hard to say what the issue might be. Feel free to stop by the `#cli` channel on [WordPress.org](https://make.wordpress.org/chat/) Slack if you'd like help debugging further.

**Comment by wojsmol** [comment]
@porg Those deprecation lines  should reference only to Request library or WordPress plugins.

**Comment by porg** [comment]
Thanks for the kind offer!
- If the next stable release update comes within a reasonable time (any ETA?),
  - I rather prefer to wait for this, and hope this "solves itself" with the regular update.
- If it then still hangs, I would very much like to take your offer and debug together in the chat.

**Comment by porg** [comment]
@wojsmol 
> Those deprecation lines should reference only to Request library or WordPress plugins.

- Someone who knows PHP certainly can make use of this your statement. 👍 
- But I only use wp-cli as a means to manage my Wordpress.
- As a developer and PHP layman I interpret your statement as: The deprecation notices only come from including/invoking a library called "Request" or certain WordPress plugins, which interact wp-cli on many calls, hence so many deprecation warning lines (b/c many calls on each wpi-cli invocation).

**Comment by wojsmol** [comment]
@porg Can you post those deprecation lines here?

**Comment by porg** [comment]
Now not package managed by brew but manually installed as `~/bin/wp` at a superficial glance gives me exactly the same deprecation line (repeated dozen times) as in my initial report (only the path of wp-cli seems to differ):

    Creation of dynamic property WP_CLI\Dispatcher\CompositeCommand::$longdesc is deprecated in phar:///Users/me/bin/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CompositeCommand.php on line 38

The full logs for you to see, maybe you note some important differences:
- [wp-info.txt](https://github.com/wp-cli/wp-cli/files/10247251/wp-info.txt)
- [wp-cli-update.txt](https://github.com/wp-cli/wp-cli/files/10247230/wp-cli-update.txt)
- [wp-cli-update-nightly.txt](https://github.com/wp-cli/wp-cli/files/10247233/wp-cli-update-nightly.txt)

**Comment by wojsmol** [comment]
@porg 
```
Creation of dynamic property WP_CLI\Dispatcher\CompositeCommand::$longdesc is deprecated in phar:///Users/me/bin/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CompositeCommand.php on line 38
```
is specific to php 8.2 and will be fixed when https://github.com/wp-cli/.github/issues/43 will be complete.

**Comment by danielbachhuber** [comment]
> If the next stable release update comes within a reasonable time (any ETA?),
> 
>     * I rather prefer to wait for this, and hope this "solves itself" with the regular update.

@porg This particular issue should "solve itself" with the next regular update. However, WP-CLI doesn't yet formally support PHP 8.2. [Stay tuned to the blog](https://make.wordpress.org/cli/) for when we have such news to announce.

**Comment by wdoekes** [comment]
Looks fixed in: https://github.com/wp-cli/wp-cli/commit/a2f0250bf85b3818ee77369af9e9d9a3d0f02cd0

**Comment by porg** [comment]
^ Thanks! Created the linked followup issue, for the next deprecation note I ran into, after the update.

**Comment by RichardTMiles** [comment]
Turn off errors from cli 
https://github.com/orgs/wp-cli/discussions/5765

**Comment by abbeymaniak** [comment]
i'm getting this error, please any fix? i'm using php 8.2.6

Deprecated: Creation of dynamic property WP_CLI\Dispatcher\Subcommand::$longdesc is deprecated in phar:///opt/homebrew/Cellar/wp-cli/2.7.1/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CompositeCommand.php on line 38
Error: Error establishing a database connection.

**Comment by shariffff** [comment]
It seems the issue was fixed here: https://github.com/wp-cli/wp-cli/pull/5707

Though latest stable version shows the same warning. But it doesn't in the nightly version. I updated it with `wp cli update --nightly`
https://wp-cli.org/#updating

**Comment by abbeymaniak** [comment]
> Stay tuned to the blog

i get the same error when i run wp cli update --nightly

**Comment by abbeymaniak** [comment]
> It seems the issue was fixed here: #5707
> 
> Though latest stable version shows the same warning. But it doesn't in the nightly version. I updated it with `wp cli update --nightly` https://wp-cli.org/#updating

Thanks for this, i had to use <code>sudo wp cli update --nightly</code> and i'm using XAMPP i had to change my cli to use xampp PHP or else i'll keep getting the error to establish database connection

**Comment by vrocks** [comment]
Do you have define( 'WP_DEBUG', false ); or do you have define( 'WP_DEBUG', true );

I have noticed that I only get this on blogs with it set to True.
